src/NUnitTestAdapter/NUnit3TestExecutor.cs

```diff
@@ -47,7 +47,8 @@ public sealed class NUnit3TestExecutor : NUnitTestAdapter, ITestExecutor, IDispo
         public void RunTests(IEnumerable<string> sources, IRunContext runContext, IFrameworkHandle frameworkHandle)
         {
 #if LAUNCHDEBUGGER
-            Debugger.Launch();
+            if (!Debugger.IsAttached)
```

Comment:
The debugger may not necessarily be attached when we reach this point. The VSTest.Executionengine is normally not started, so there is nothing to attach to.  For the discoveryengine, that one is normally running after the first test run, but I am in doubt whether it should check there too.